### ISSUE TITLE
Halpe dataset (136 keypoints) pre-trained model

### ISSUE THREAD

**Issue by NeelayS** [MVIG-SJTU/AlphaPose#695]
Hi, any estimate regarding till when the pre-trained weights for the Fast Pose model with 136 keypoints prediction will be available ?

**Comment by Fang-Haoshu** [comment]
You may try the Halpe 136 model here: https://drive.google.com/file/d/1NyOuVLb5oIYP4LZmp9BSAQdbGjEk--gv/view?usp=sharing
cfg file: https://drive.google.com/file/d/1d9BSHnIXS3OVGNa4lc1de4qTsybY-N9_/view?usp=sharing

Remember to  modify the num_joints from 133 to 136.

But note that it is not the final model. We are still training it so far and could expect a better accuracy afterwards. Stay tuned!

**Comment by NeelayS** [comment]
Thanks a lot!

**Comment by Fang-Haoshu** [comment]
Enjoy!

**Comment by Fang-Haoshu** [comment]
Oh, another thing is that the score may be small for visualization. You can multiple the score for 2 times.

**Comment by maxzuo** [comment]
Hi there, I'm having an error using the cfg file provided above. I get this error:

```
Traceback (most recent call last):
  File "scripts/demo_inference.py", line 175, in <module>
    det_loader = DetectionLoader(input_source, get_detector(args), cfg, args, batchSize=args.detbatch, mode=mode, queueSize=args.qsize)
  File "/AlphaPose/alphapose/utils/detector.py", line 49, in __init__
    pose_dataset = builder.retrieve_dataset(self.cfg.DATASET.TRAIN)
  File "/AlphaPose/alphapose/models/builder.py", line 45, in retrieve_dataset
    exec(f'from ..datasets import {cfg.TYPE}')
  File "<string>", line 1, in <module>
ImportError: cannot import name 'coco_wholebody'
```

Could you help me fix this? Thanks.

**Comment by yahahahoo** [comment]
> 嗨，我在使用上面提供的cfg文件时出错。我收到此错误：
> 
> ```
> Traceback (most recent call last):
>   File "scripts/demo_inference.py", line 175, in <module>
>     det_loader = DetectionLoader(input_source, get_detector(args), cfg, args, batchSize=args.detbatch, mode=mode, queueSize=args.qsize)
>   File "/AlphaPose/alphapose/utils/detector.py", line 49, in __init__
>     pose_dataset = builder.retrieve_dataset(self.cfg.DATASET.TRAIN)
>   File "/AlphaPose/alphapose/models/builder.py", line 45, in retrieve_dataset
>     exec(f'from ..datasets import {cfg.TYPE}')
>   File "<string>", line 1, in <module>
> ImportError: cannot import name 'coco_wholebody'
> ```
> 
> 你能帮我解决这个问题吗？谢谢。

我也是

**Comment by nikhilweee** [comment]
@maxzuo I changed all occurences of `coco_wholebody` to `Halpe_136` in the YAML file in addition to modifying `num_joints` to 136 as suggested by @Fang-Haoshu and it works.

**Comment by yahahahoo** [comment]
> @maxzuo I changed all occurences of `coco_wholebody` to `Halpe_136` in the YAML file in addition to modifying `num_joints` to 136 as suggested by @Fang-Haoshu and it works.

But it seems that it will report an error when it uses the validation set

**Comment by maxzuo** [comment]
@nikhilweee thank you! the demo script works now, though visually when I check the saved results, it doesn't seem to be detecting whole bodies, only parts like an arm or the face... do you know if this is an issue with the visualization process or if it is an issue with the model itself? Thanks again mate.

**Comment by nikhilweee** [comment]
@maxzuo I also face the same issue. Not all parts of the body are detected. I am tempted to think that this is because the model in https://github.com/MVIG-SJTU/AlphaPose/issues/695#issuecomment-695879500 is still not final. Although @Fang-Haoshu would be the best person to comment.

**Comment by nikhilweee** [comment]
@yahahahoo You might be right. I'm not deeply familiar with the repository. I just happened to do something and it worked.

**Comment by yahahahoo** [comment]
> @maxzuo I also face the same issue. Not all parts of the body are detected. I am tempted to think that this is because the model in [#695 (comment)](https://github.com/MVIG-SJTU/AlphaPose/issues/695#issuecomment-695879500) is still not final. Although @Fang-Haoshu would be the best person to comment.

You can refer to this url to train a model with 136 key points：https://github.com/Fang-Haoshu/Halpe-FullBody
I think the effect is pretty good

**Comment by li2508264** [comment]
Oh,there are 204 points in the result json, not 136?

**Comment by bshreyas13** [comment]
Hey guys I am trying to train on Halpe_136 from scratch, and it crashes when it starts training and tries to pull the img_id in custom.py.. I have extracted the images in DATA1/Benchmark/... as per the path in the script..
My annotations file is in data/halpe_136/annotations and I have pointed to this path in the config file.

I am not sure my it's getting a dict for img_path instead of the path to the image. Any inputs on setting up halpe_136 and training it would help.

Thanks
